### PR TITLE
Adding db cleanup migrations

### PR DESCRIPTION
**Pull request by Poltergeistz**
# Description

DL-5601

This PR adds new migrations for the db cleanup service. See changes in https://github.com/lblod/db-cleanup-service/pull/1

After testing locally jobs are running and resources are successfully deleted but there is cases where a job has a lot of resources, thinking about this culprit <http://lblod.data.gift/id/jobs/concept/JobOperation/downloadUrlWarning>
```ttl
<http://data.lblod.info/id/cleanup-job/3f623084-416f-44ef-90fa-7c7fa3d4efad> a cleanup:Job ;
    mu:uuid "3f623084-416f-44ef-90fa-7c7fa3d4efad" ;
    dcterms:title "clean up more than a month old jobs and tasks from harvesting graph" ;
    cleanup:selectPattern """
              GRAPH <http://mu.semte.ch/graphs/harvesting> {
                VALUES ?operation {
                  <http://lblod.data.gift/id/jobs/concept/JobOperation/berichtencentrumWarning>
                  <http://lblod.data.gift/id/jobs/concept/JobOperation/downloadUrlWarning>
                }
                ?resource a <http://vocab.deri.ie/cogs#Job> ;
                  <http://redpencil.data.gift/vocabularies/tasks/operation> ?operation;
                  <http://purl.org/dc/terms/modified> ?modified .
                FILTER (?modified < (NOW() - "P1M"^^xsd:duration))
                ?task <http://purl.org/dc/terms/isPartOf> ?resource.
                ?resource ?p ?o.
                ?task ?taskP ?taskO.
                OPTIONAL {
                  ?task <http://redpencil.data.gift/vocabularies/tasks/resultsContainer> ?resultsContainer .
                  ?resultsContainer ?presultsContainer ?oresultsContainer .
                }
                OPTIONAL {
                  ?task <http://redpencil.data.gift/vocabularies/tasks/inputContainer> ?inputContainer .
                  ?inputContainer ?pinputContainer ?oinputContainer .
                }
              }
              """ ;
    cleanup:deletePattern """
             GRAPH <http://mu.semte.ch/graphs/harvesting> {
               ?resource ?p ?o.
               ?task ?taskP ?taskO.
               ?resultsContainer ?presultsContainer ?oresultsContainer .
               ?inputContainer ?pinputContainer ?oinputContainer .
             }
              """ ;
    cleanup:cronPattern "*/45 * * * *" . 
``` 
As an indication, there are more than a million resources to be deleted locally, which may take some time until all the operations are completed, so they should be spaced out over time with other resource-consuming operations.

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

# Related services

- N/A

# How to test

:warning: **Important:  Make sure to make a backup of your data to test this feature**

1. Link PR volume or use image in loket
2. Add cleanup migrations through Loket PR (link pr here)
3. Let the service run it should delete resources at configured cronPattern
 
# What to check

- Decide which cronPattern might fit for each jobs

# Links to other PR's

- https://github.com/lblod/db-cleanup-service/pull/1

# Notes

- N/A

### CODE REVIEW

File: config/migrations/2024/20240620110047-cleanup-service-migrations/20240620110047-add-errors-cleanup.ttl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+              ?resource a <http://mu.semte.ch/vocabularies/ext/KalliopeSyncError> ;
+                <http://purl.org/pav/createdOn> ?createdOn .
+              FILTER (?createdOn < (NOW() - "P1M"^^xsd:duration))
+              ?resource ?p ?o . }
+              """ ;
+    cleanup:deletePattern """
+              GRAPH <http://mu.semte.ch/graphs/public> {
+              ?resource ?p ?o . }
+              """ ;
+    cleanup:cronPattern "*/10 * * * *" .
```
Comment (Poltergeistz) [inline config/migrations/2024/20240620110047-cleanup-service-migrations/20240620110047-add-errors-cleanup.ttl:19]: Note that this is just for testing purpose and should be modified.

File: config/migrations/2024/20240620110047-cleanup-service-migrations/20240620110047-add-errors-cleanup.ttl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+@prefix dcterms: <http://purl.org/dc/terms/> .
+
+<http://data.lblod.info/id/cleanup-job/d4318463-4cd0-40c2-b67a-5e29d06db2d8> a cleanup:Job ;
+    mu:uuid "d4318463-4cd0-40c2-b67a-5e29d06db2d8" ;
+    dcterms:title "clean up more than a month old Kalliope errors" ;
+    cleanup:selectPattern """
+              GRAPH <http://mu.semte.ch/graphs/public> {
+              ?resource a <http://mu.semte.ch/vocabularies/ext/KalliopeSyncError> ;
+                <http://purl.org/pav/createdOn> ?createdOn .
+              FILTER (?createdOn < (NOW() - "P1M"^^xsd:duration))
```
[the remark this answers is not in the document]
Comment (Riadabd) [reply config/migrations/2024/20240620110047-cleanup-service-migrations/20240620110047-add-errors-cleanup.ttl:12]: I think a month is good. Should there be any need to investigate some error that happened a while back, we can easily use the backups for that.

File: config/migrations/2024/20240620110047-cleanup-service-migrations/20240620110047-add-errors-cleanup.ttl [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                ))
+                FILTER (?created < (NOW() - "P1M"^^xsd:duration))
+                ?resource ?p ?o .
+              }
+              """ ;
+    cleanup:deletePattern """
+              GRAPH ?g {
+              ?resource ?p ?o . }
+              """ ;
+    cleanup:cronPattern "*/25 * * * *" .
```
Comment (Riadabd) [inline config/migrations/2024/20240620110047-cleanup-service-migrations/20240620110047-add-errors-cleanup.ttl:54]: This needs an end of file, to be similar as the other ttl file.

**Comment by Riadabd** [comment]
There is one remaining cleanup job (`http://data.lblod.info/id/cleanup-job/10724bc2-c9d0-4a35-a499-91a8b7cb023b`) that needs to be updated. It is still using a number for the date calculations in the `FILTER` instead of using the new preferred method (using `BIND(NOW() - xsd:dayTimeDuration("P1D") AS ?oneDayAgo)`).

File: config/migrations/2024/20240620110047-cleanup-service-migrations/20240620110047-add-errors-cleanup.ttl (hunk 3 above)
Comment (Poltergeistz) [reply config/migrations/2024/20240620110047-cleanup-service-migrations/20240620110047-add-errors-cleanup.ttl:54]: You mean a new line ?

**Pull request merged**